wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -146,18 +154,32 @@ class WirePlugin @Inject constructor(
     }
 
     javaTarget?.let {
-      val compileTask = project.tasks.named("compileJava") as TaskProvider<JavaCompile>
-      compileTask.configure {
+      val compileJavaTask = project.tasks.named("compileJava") as TaskProvider<JavaCompile>
+      compileJavaTask.configure {
         it.source(javaOutDirs)
         it.dependsOn(wireTask)
       }
+      if (kotlin) {
+        sourceSetContainer = project.property("sourceSets") as SourceSetContainer
+        val mainSourceSet = sourceSetContainer.getByName("main") as SourceSet
+        mainSourceSet.java.srcDirs(javaOutDirs)
+
+        val compileKotlinTask = project.tasks.named("compileKotlin") as TaskProvider<KotlinCompile>
+        compileKotlinTask.configure {
+          it.dependsOn(wireTask)
+        }
+      }
     }
 
     kotlinTarget?.let {
-      val compileTask = project.tasks.named("compileKotlin") as TaskProvider<KotlinCompile>
-      compileTask.configure {
-        it.source(kotlinOutDirs)
-        it.dependsOn(wireTask)
+      try {
+        val compileKotlinTask = project.tasks.named("compileKotlin") as TaskProvider<KotlinCompile>
+        compileKotlinTask.configure {
+          it.source(kotlinOutDirs)
+          it.dependsOn(wireTask)
+        }
+      } catch (e: UnknownDomainObjectException) {
```

Comment:
Is there a way to query this that doesn't involve catching an exception?

Reply:
not in TaskProvider world, the others would `realize` and return the actual Task.

Reply:
That seems like a really weird ergonomic issue. Should we file a Gradle bug asking for a `configureOrElse` or something?

Reply:
You're referring to https://github.com/gradle/gradle/issues/8057, and it's a whole can of worms. Your options are either try/catch or `findByName`, which will realize the task.

Reply:
Thanks @autonomousapps, maybe I'll employ some of the suggestions in gradle/gradle#8057.

Reply:
I guess it's not clear why try/catch, a synchronous construct, is able to work where a query or some other mechanism wouldn't.

Reply:
I don't disagree with you, and I welcome any input you might have on the issue.

Reply:
I wonder if returning a NamedDomainObjectCollection would be better, since then we could configureEach which no-ops if the task doesn't exist?

Reply:
`tasks.withType<KotlinCompile>().configureEach { ... }` would lazily configure all KotlinCompile tasks (and not failing if one weren't found).